### DISCUSSION TITLE
How to Encrypt/Decrypt PII Data

### DISCUSSION THREAD

**Discussion by shiva-rakshith** [Sunbird-RC/community#779] [General]
There are few fields like email address and mobile number in our schema, which are PII data. 

Do we have an option to encrypt and decrypt the PII fields in the registry?

**Comment by holashchand** [comment]
Hey @shiva-rakshith, encryption/decryption of any fields is not yet part of registry.

But this would be a nice to have feature, Do you have any thoughts on how do you want to configure PII fields in registry schema?

**Reply by shiva-rakshith** [reply]
Hi @holashchand 

In _osConfig, we can define `encryptionFields` and add the fields to be encrypted to it. The registry should encrypt and store these fields in the database.

```
_osConfig: {
     "encryptionFields": ["email", "mobile"]
}
```

**Reply by holashchand** [reply]
Hi @shiva-rakshith, We have picked up this task. 

Here is the issue created https://github.com/Sunbird-RC/community/issues/798

Their are some open questions in there, It would be great if you could comment on those
